subprojects/language-java/src/integTest/groovy/org/gradle/javadoc/JavadocIntegrationTest.groovy

```diff
@@ -61,6 +61,20 @@ class JavadocIntegrationTest extends AbstractIntegrationSpec {
         file("build/docs/javadoc/Foo.html").text.contains("""Hey Joe!""")
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/1090")
+    def "escape single quotes in javadoc options"() {
+        buildFile << """
+            apply plugin: "java"
+            javadoc.options.header = "where you goin' with that gun in your hand"
+        """
+
+        file("src/main/java/Foo.java") << "public class Foo {}"
+
+        when: run("javadoc", "-i")
+        then:
```

Comment:
Let's have a single test (the logic will be the same) with a `where` clause that tests different inputs. When you change the test please also add the `@Unroll` annotation and indicate some sort of description in the test method name. You will find other examples in the Gradle core code base.